### ISSUE TITLE
[RAC] Define Alerts DataGrid cell action popover behaviors

### ISSUE THREAD

**Issue by mdefazio** [elastic/stack-design-team#85]
## Problem:
There are currently the same cell actions for every cell and will at times overlap to the following cell making it impossible to view the cell's content.

![image](https://user-images.githubusercontent.com/3756330/129238887-640eeb57-f688-405c-a4b5-d3ec47ab77b8.png)

## Solution:
We need to define specific cell actions for each (default) cell and then the default actions for new cells. 

- [x] Show cell actions for default cells in Alerts Table
- [x] Show cell actions for all other cells in Alerts Table
- [x] Show how the event renderer will appear (and reason statement)
- [x] Create observability issue for approved updates
- [x] Create security issue for approved updates

-----

### Observability

Mockup:
![image](https://user-images.githubusercontent.com/3756330/130522806-1bbe263f-26be-4965-841e-41af4f05af34.png)

We will limit the number of exposed actions on cell hover and rely on the popover to house additional actions:

1. Severity and Reason are the only default columns with hover/expand action
2. Reason popover provides primary action to investigate and view alert summary
3. Severity popover provides only Filter in / Filter out actions (no labels, only tooltips)
4. Additional fields will default to provide an expand action with Filter In / Out in popover
5. No click events on the cell values themselves

| Cell | Action | Popover |
| :-- | :-- | :-- |
| Timestamp | No expand or click event |  n/a |
| Alert status | No expand or click event | n/a |
| Duration | No expand or click event | n/a |
| Severity | Expand action, no click event | Filter In / Out |
| Reason | Expand action, no click event | Investigate in app, View alert summary |
| **Default for added fields** | Expand action, no click event | Cell value with Filter In/Out |

### Security

Mockup:
![image](https://user-images.githubusercontent.com/3756330/130636295-5e086ee5-7c0e-45ac-892c-3ca06fdbda19.png)

We will limit the number of exposed actions on cell hover and rely on the popover to house additional actions:

1. We should avoid click events on the cell value and rely on the popover instead
2. Security workflow should allow for advanced user to quickly add to timeline.  So default hover options are ‘Add to timeline’ and ‘Expand’
3. Default set of actions in cell popover are: Filter In/Out, Copy (includes field key) and Investigate in Timeline
4. Rule popover will provide additional ‘View rule’ action
5. Host popover will provide additional ‘View host’ action
6. Max width for text in popover is 360px
7. Reason field includes Event renderer in popover.  Needs a max-width to avoid going full screen width (suggest 60% of Alert table width)

Cells that should **NOT** have cell actions: 
- Risk score
- Timestamp

**Reason field** does NOT include the timeline cell action

Top Values may also be necessary to include in popover (in discussion—Would still recommend hiding by default / needing a click to show.  Example in mockup)

**Comment by ryankeairns** [comment]
Since #108891 has been closed with a partial fix (overflow hidden), we should probably create a new issue to track the number of cell actions. #108921 is related but focuses on the number actions in the popover panel, specifically. Ideally those are fixed together, however, engineering was going to attempt a minimum column width as another short-term fix for 7.15. 

Now that I've written that thought out, I'll go ahead and creat a new issue focused on that alone. It's a hacky, short term solution, but we ought to at least track it.

**Comment by mdefazio** [comment]
EUI discussion has been answered and a issue filed in their repo: https://github.com/elastic/eui/issues/5132

**Security specific proposed implementation:**
Two options provided below.  Decision is handed to Security team to decide which makes the most sense

![image](https://user-images.githubusercontent.com/3756330/131888388-3147510a-929d-4a24-a9f6-fa8fd429922a.png)

- - -

**Observability specific proposed implementation:**
![image](https://user-images.githubusercontent.com/3756330/131889388-322317a8-0b27-42bb-800d-7e9f0d21f4d7.png)

@katrin-freihofner Can you confirm the Observability proposal?  I think these are likely ready to have an engineering ticket created (I'm happy to do the Security one)

**Comment by katrin-freihofner** [comment]
Thank you @mdefazio I created [this ticket](https://github.com/elastic/kibana/issues/111090) for Observability.

**Comment by mdefazio** [comment]
Had a conversation with Paul, Adam and Xavier and Security will go with Filter as the priority.  A custom popover is still going to be needed and so Security shouldn't be blocked by any EUI work.  Security engineering issue coming soon.

**Comment by mdefazio** [comment]
Security issue to update cell actions: https://github.com/elastic/kibana/issues/112581
